src/test/java/com/mongodb/InterruptibilityItTest.java

```diff
@@ -191,15 +202,21 @@ private static Stream<Arguments> arguments() {
     private static <T extends Throwable> T assertThrowsInVirtualInterruptedThread(
             final Class<T> expectedType,
             final Duration interruptDelay,
-            @Nullable final String blockedCommandName,
-            final Runnable action) {
-        return withBlockedCommand(blockedCommandName, blockTimeout, () ->
-                assertThrows(expectedType, () ->
-                        inVirtualThread(() ->
-                                inInterruptedThread(interruptDelay, action)
-                        )
-                )
-        );
+            final String commandName,
+            final boolean blockCommand,
+            final Runnable executeCommand) {
+        return withBlockedCommand(blockCommand ? commandName : null, blockTimeout, () -> {
+            startedCommandNames.clear();
+            T t = assertThrows(expectedType, () ->
+                    inVirtualThread(() ->
+                            inInterruptedThread(interruptDelay, executeCommand)
+                    )
+            );
+            assertFalse(startedCommandNames.stream().anyMatch(startedCommandName -> !startedCommandName.equals(commandName)),
+                    "Unexpected commands in " + startedCommandNames + ", expected only " + commandName);
+            assertTrue(startedCommandNames.size() <= 1, startedCommandNames + "was attempted " + startedCommandNames.size() + " times");
```

Comment:
`commandName` is never `null` here. The reason `<= 1` is used is that an operation may be interrupted either before `CommandStartedEvent`, or after.

Reply:
Ah, ok.  But still, doesn't each test know the expected number of commands that will be started, e.g. the server selection test expects 0, and so on?  If so, each test could pass in the expected number of commands. 

If you think it's overkill, I'll let it go.  I point it out only because I have vague memories of being burned by fuzzy assertions like this.